### PR TITLE
Homework4: Renat Levotskiy

### PR DESCRIPTION
**Pull request by renat33**

### CODE REVIEW

File: hw4_Zadacha2.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Дан список стран и городов каждой страны.
```
Comment (belovmd) [inline hw4_Zadacha2.py]: what about brest in belarus and france?

File: hw4_Zadacha5.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# Каждый из N школьников некоторой школы знает Mi языков.
+# Определите, какие языки знают все школьники и языки,
+# которые знает хотя бы один из школьников.
+
+pupils = int(input())
```
Comment (belovmd) [inline hw4_Zadacha5.py]: please add input invite string

File: hw4_Zadacha5.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+lang_list1 = []
+all_lang = set()
+
+for i in range(pupils):
+    lang = int(input())
+    lang_list = set()
+    lang_list1.append(lang_list)
+    for u in range(lang):
+        q = input()
```
Comment (belovmd) [inline hw4_Zadacha5.py]: what is the q and u and where we overall use u? 
for _ in ...

File: hw4_Zadacha5.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        lang_list.add(q)
+        all_lang.add(q)
+
+all_pup = lang_list1[0]
```
Comment (belovmd) [inline hw4_Zadacha5.py]: all_pup - unclear naming

File: hw4_Zadacha7.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+num1 = int(input())
+num2 = int(input())
+while num1 != 0 and num2 != 0:
+    if num1 > num2:
+        num1 = num1 % num2
+    else:
+        num2 = num2 % num1
+
+print(num1 + num2)
```
Comment (belovmd) [inline hw4_Zadacha7.py]: it's a hack. Algorithm doesn't expect result as a sum. Here you can just try not to detect 0 in one of your variables

**Review by belovmd — CHANGES_REQUESTED** [review]
Please take a look at the next comments

File: hw4_Zadacha2.py (hunk 1 above)
Comment (renat33) [reply hw4_Zadacha2.py]: fixed

File: hw4_Zadacha5.py (hunk 2 above)
Comment (renat33) [reply hw4_Zadacha5.py]: fixed

File: hw4_Zadacha5.py (hunk 3 above)
Comment (renat33) [reply hw4_Zadacha5.py]: fixed

File: hw4_Zadacha5.py (hunk 4 above)
Comment (renat33) [reply hw4_Zadacha5.py]: fixed

File: hw4_Zadacha7.py (hunk 5 above)
Comment (renat33) [reply hw4_Zadacha7.py]: fixed?

**Review by belovmd — COMMENTED** [review]
9

**Pull request closed without merging**